modules/m_whois.c

```diff
@@ -269,29 +266,38 @@ single_whois(struct Client *source_p, struct Client *target_p, int operspy)
 
 	if (!IsService(target_p))
 	{
-		RB_DLINK_FOREACH(ptr, target_p->user->channel.head)
+		hook_data_channel_visibility hdata_vis;
+		rb_dlink_node *ps, *pt;
+		struct Channel *chptr;
+		struct membership *ms, *mt;
+
+		hdata_vis.client = source_p;
```

Comment:
If we add the target to the hook struct:

```suggestion
		hdata_vis.client = source_p;
		hdata_vis.target = target_p;
```